agent/stats/reporter/reporter.go

```diff
@@ -99,18 +109,23 @@ func NewDockerTelemetrySession(
 func (session *DockerTelemetrySession) Start(ctx context.Context) error {
 	backoff := retry.NewExponentialBackoff(time.Second, 1*time.Minute, 0.2, 2)
 	for {
+		select {
+		case <-ctx.Done():
+			logger.Info("TCS session exited cleanly.")
```

Comment:
nit & non-blocking. Maybe we should use `ECS Telemetry service (TCS)` in log messages for consistency.

Reply:
Will update. I think we have this in ACS logging too. We can change it in ACS changes.